projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx

```diff
@@ -10,6 +10,11 @@ const CLOSE_TO_LIMIT_PERCENT = 0.8;
 const DISMISSED_NOTICES = 'jetpack-search-dismissed-notices';
 
 const getNotices = ( tierMaximumRecords = null ) => {
+	const recordLimit =
+		typeof tierMaximumRecords === 'number'
+			? tierMaximumRecords?.toLocaleString()
```

Comment:
If we reach this, `tierMaximumRecords` has to not be null or undefined. Either of those would have returned a different value from the `typeof` test.
```suggestion
			? tierMaximumRecords.toLocaleString()
```
I'm guessing you copied this from similar code in `projects/packages/search/src/dashboard/components/record-meter/record-count.jsx`, which it looks like did like `tierMaximumRecords?.toLocaleString()` in 75f7bf71a713fca09409eafb0650eb3eb200218e and then added the typeof check in ed227e6f0c9e41abbbe10ccbfbd6a26b52388ad1 without dropping the now-unneeded `?`.